src/systems/sensors/Sensors.cc

```diff
@@ -210,7 +210,7 @@ void SensorsPrivate::WaitForInit()
         this->renderUtil.SetBackgroundColor(*this->backgroundColor);
       if (this->ambientLight)
         this->renderUtil.SetAmbientLight(*this->ambientLight);
-      this->renderUtil.Init();
```

Comment:
the problem for me was that the RenderUtil /ogre initialization is now moved to a different thread causing subsequent render calls to segfault. Interesting that it's not an issue on macOS. Perhaps we can ifdef this?

Reply:
That is interesting - there may be an issue on Metal but I'm not seeing side affects yet. For OpenGL it was using the OpenGLContext on mismatched threads that caused problems in `MinimalScene`. I see there's a comment in `RenderUtil.hh` that `Init` must be called on the render thread, but that will break on macOS when Ogre asks for a new render window (both Metal and OpenGL). In the initialisation code for `MinimalScene` there is a bit of juggling where the context and render thread are moved from the QSGRenderThread to the Main thread for initialisation, then to the RenderThread once that's complete. A similar approach may be needed here (or an #ifdef as you suggest!)

Reply:
Added #ifdef in [81337d2](https://github.com/gazebosim/gz-sim/pull/1225/commits/81337d2a1f9e5a488077d34c1b4a69dac174ec4e) to leave behaviour unchanged if not macOS.